src/main/java/com/kenya/jug/arena/controller/AuthController.java

```diff
@@ -0,0 +1,73 @@
+package com.kenya.jug.arena.controller;
+
+import com.kenya.jug.arena.io.AuthRequest;
+import com.kenya.jug.arena.io.AuthResponse;
+import com.kenya.jug.arena.service.AppUserDetailsService;
+import com.kenya.jug.arena.service.UserService;
+import com.kenya.jug.arena.util.JwtUtil;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseCookie;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.DisabledException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+
+@RestController
+@RequiredArgsConstructor
+public class AuthController {
+
+    private final AuthenticationManager authenticationManager;
+    private final AppUserDetailsService appUserDetailsService;
+    private final JwtUtil jwtUtil;
+    private final UserService userService;
+
+    @PostMapping("/login")
+    public ResponseEntity<?> login(@Valid @RequestBody AuthRequest request) {
+        try {
+            authenticate(request.getEmail(), request.getPassword());
+            final UserDetails userDetails = appUserDetailsService.loadUserByUsername(request.getEmail());
+            final String jwtToken = jwtUtil.generateToken(userDetails);
+
+            ResponseCookie cookie = ResponseCookie.from("jwt", jwtToken)
+                    .httpOnly(true)
+                    .path("/")
+                    .maxAge(Duration.ofDays(1))
+                    .sameSite("Strict")
+                    .build();
+            return ResponseEntity.ok().header(HttpHeaders.SET_COOKIE, cookie.toString()).body(new AuthResponse(request.getEmail(), jwtToken));
+        } catch (BadCredentialsException exception) {
```

Comment:
Consider replacing this error handling with a dynamic RestControllerAdvice AOP Class, should result in cleaner code